src/methods/polygonize.jl

```diff
@@ -56,166 +50,180 @@ f
+translating to inside or outside the polygons.
 
 If `xs` and `ys` are passed in they are used as the pixel center points.
 
+
 # Keywords
-- `minpoints`: ignore polygons with less than `minpoints` points. 
+
+- `minpoints`: ignore polygons with less than `minpoints` points.
+
+# Example
+
+```julia
+using GeometryOps
+x = polygonize(>(0.6), rand(100, 100), minpoints=3); 
+using GLMakie
+using Shapefile
+Makie.plot(x)
+```
 """
-polygonize(A::AbstractMatrix; kw...) = polygonize(axes(A)..., A; kw...) 
-
-function polygonize(xs, ys, A::AbstractMatrix; minpoints=10)
-    ## This function uses a lazy map to get contours.  
-    contours = Iterators.map(get_contours(A)) do contour
-        poly = map(contour) do xy
-            x, y = Tuple(xy)
-            Point2f(xs[x], ys[y])
-        end
+polygonize(A::AbstractMatrix{Bool}; kw...) = polygonize(identity, A; kw...)
+polygonize(f::Base.Callable, A::AbstractMatrix; kw...) = polygonize(f, axes(A)..., A; kw...)
+polygonize(xs::AbstractRange, ys::AbstractRange, A::AbstractMatrix{Bool}; kw...) =
+    polygonize(identity, xs, ys, A)
+function polygonize(f::Base.Callable, xs::AbstractRange, ys::AbstractRange, A::AbstractMatrix; 
+    kw...
+)
+    # Make vectors of pixel bounds
+    xhalf = step(xs) / 2
+    yhalf = step(ys) / 2
+    # Make bounds ranges first to avoid floating point error making gaps or overlaps
+    xbounds = first(xs) - xhalf : step(xs) : last(xs) + xhalf
+    ybounds = first(ys) - yhalf : step(ys) : last(ys) + yhalf
+    Tx = eltype(xbounds)
+    Ty = eltype(ybounds)
+    xvec = Vector{Tuple{Tx,Tx}}(undef, length(xs))
+    yvec = Vector{Tuple{Ty,Ty}}(undef, length(ys))
+    for i in eachindex(xvec)
+        xvec[i] = xbounds[i], xbounds[i+1]
     end
-    ## If we filter off the minimum points, then it's a hair more efficient
-    ## not to convert contours with length < missingpoints to polygons.
-    if minpoints > 1
-        contours = Iterators.filter(contours) do contour
-            length(contour) > minpoints
-        end
-       return map(Polygon, contours)
-    else
-        return map(Polygon, contours)
+    for i in eachindex(yvec)
+        yvec[i] = ybounds[i], ybounds[i+1]
     end
+    return _polygonize(f, xvec, yvec, A; kw...)
 end
-
-## rotate direction clockwise
-rot_clockwise(dir) = (dir) % 8 + 1
-## rotate direction counterclockwise
-rot_counterclockwise(dir) = (dir + 6) % 8 + 1
-
-## move from current pixel to next in given direction
-function move(pixel, image, dir, dir_delta)
-    newp = pixel + dir_delta[dir]
-    height, width = size(image)
-    if (0 < newp[1] <= height) && (0 < newp[2] <= width)
-        if image[newp] != 0
-            return newp
+polygonize(f::Base.Callable, xs::AbstractVector, ys::AbstractVector, A::AbstractMatrix; kw...) =
+    _polygonize(f, xs, ys, A; kw...)
+
+function _polygonize(f, xs::AbstractVector{T}, ys::AbstractVector{T}, A::AbstractMatrix; 
+    minpoints=10,
```

Comment:
What's the motivation for this default?

Reply:
Yeah, good question. I guess it should be zero.